code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java

```diff
@@ -231,18 +231,7 @@ public boolean readyForEvent(@NonNull final Event event) {
             }
         }
 
-        if (!hasValidSharedState(
-                IdentityConstants.EventDataKeys.Configuration.MODULE_NAME, event)) {
-            Log.trace(
-                    IdentityConstants.LOG_TAG,
-                    LOG_SOURCE,
-                    "Waiting for the Configuration shared state to get required configuration"
-                            + " fields before processing [event: %s].",
-                    event.getName());
-            return false;
-        } else {
-            return true;
-        }
+        return true;
```

Comment:
The check for a valid Configuration is still needed for the block above (lines 213-232) for appendToUrl and getUrlVariables. The logic for that block relied on the Configuration state check if the block evaluated to true and did not return false due to the Analytics state check. If the current Configuration state is not set then appendToUrl/getUrlVariables won't necessarily fail, but they may not include the org ID in their responses.

Reply:
I can add an explicit configuration check but the behaviour now is same as before. The `forceSyncIdentifiers` call only succeeds if the SDK has a valid configuration that includes an org-id. Right now, we only check the shared state in the context of the specific event, and the call is assumed to always pass if `forceSyncIdentifiers` succeeds. In this particular case, we had a valid (but empty) configuration, which will allow the call to proceed, but the org-id was still missing.

I think we should reconsider the broader approach of how we validate states in `readyForEvent`.

Reply:
Okay, I see your point. Adding this check back for configuration is just checking the configuration wasn't cleared which isn't really possible, so it's not necessary. Sorry for the confusion, you can roll back your last commit if you want.